### DISCUSSION TITLE
How efficiently processing the result from `str.split`?

### DISCUSSION THREAD

**Discussion by yohplala** [vaexio/vaex#1909] [Q&A]
Hi,

This question is for my own information, i.e. it is a 'general' question.

I was recently answered in #1906 by Jovan that following approach to process results from `str.split` in vaex is not efficient (use of `apply` ).
```python
max_date = vdf_max_date["fnames"].str.split('.').str.split('-',1).apply(lambda x: np.datetime64(x[0][1])).max().item()
```
(just for being complete, this methodology comes [from this SO answer](https://stackoverflow.com/a/67600269/4442753))

Meanwhile, I also stumbled on [this Medium article](https://towardsdatascience.com/vaex-a-dataframe-with-super-strings-789b92e8d861), showcasing speed of vaex for operations on string, including `split`.

So, the underlying thought I have is that, if vaex shows amazing speed for `split`, surely, the way of efficiently processing results from `split` has also been thought about.
Hence my question, what kind of processings ('efficient' ones) can be done after a `split`?

`split` produces an expression which items are lists.
My need would be a way to select an item in these lists.
This might not be the 'usual' processing done after a `split`. If this is the case indeed, please, what processing after a `split` can be managed efficiently?

Thanks in advance for your feedbacks!
Bests

**Comment by maartenbreddels** [comment]
I think the only thing we currently support is joining back, note that you can operate on a list of string as if it is a flattened column of strings. so you can do e.g. split -> strip -> join
